docs/develop/understanding-the-javascript-api-for-office.md

```diff
@@ -28,31 +28,89 @@ For more details around the Office.js CDN, including how versioning and backward
+
+- The `Office.onReady` event has a default handler that returns a Promise object. This gives you an alternative way of responding to the event. See **Initialize with Office.onReady** below.
+
+- Your code *must* assign some handler to `Office.initialize` although it can be a function that does nothing. See **Initialize with Office.initialize** below. You are not required to assign anything to `Office.onReady`.
+
+For more details about the sequence of events when an add-in is initialized, see [Loading the DOM and runtime environment](loading-the-dom-and-runtime-environment.md).
+
+### Initialize with Office.onReady
+
+`Office.onReady` fires first when the Office.js library is fully loaded and ready for user interaction. You can use the `Office.onReady` event handler to implement common add-in initialization scenarios, such as prompting the user to select some cells in Excel, and then inserting a chart initialized with those selected values. You can also use the handler to initialize other custom logic for your add-in, such as establishing bindings, prompting for default add-in settings values, and so on. For example, the following handler checks to see that the user's version of Excel supports all the APIs that the add-in might call.
+
+```js
+Office.onReady = function () {
+    if (!Office.context.requirements.isSetSupported('ExcelApi', '1.7')) {
+        console.log("Sorry, this add-in only works with newer versions of Excel.");
+    }
+};
+```
+
+If you are using additional JavaScript frameworks that include their own initialization handler or tests, these should be placed within the `Office.onReady` event. For example, [JQuery's](https://jquery.com) `$(document).ready()` function would be referenced as follows:
+
+```js
+Office.onReady = function () {
+    // Office is ready
+    $(document).ready(function () {
+        // The document is ready
+    });
+  };
+```
+
+If you prefer, you can call the default handler for `Office.onReady` and respond to the event when the Promise is resolved. This is the syntax in JavaScript:
+
+```js
+Office.onReady()
+    .then(function() {
+        console.log("Office is now ready");
+    });
+```
+
+Here is the same example using TypeScript:
+
+```ts
+(async () => {
+    await Office.onReady();
+    console.log("Office is now ready!");
+})();
+```
+
+> [!NOTE]
+> Even if you use `Office.onReady`, you must still assign some function to  `Office.initialize`, even if it is a function that does nothing. See **Initialize with Office.initialize** below. 
```

Comment:
No, you can now leave Office.initialize off.  Going forward, no one needs to use Office.initialize